src/server/tasks/queue-autosend-initials.ts

```diff
@@ -168,45 +171,57 @@ export const queueAutoSendOrganizationInitials: Task = async (
     [contactsToQueueInOneMinute, autosendingMps, organizationId]
   );
 
-  const { rows: totalCountToSend } = await helpers.query<{
-    campaign_id: string;
-    total_count_to_send: string;
-  }>(
+  const campaignIdsQueued = contactsQueued.map((cq) => cq.campaign_id);
+
+  await helpers.query(
     `
-      select cc.campaign_id, count(*) as total_count_to_send
-      from campaign_contact cc
-      join campaign c on cc.campaign_id = c.id
+      with sendable_contacts as (
+        select id, campaign_id
+        from campaign_contact
+        where true
+          and archived = false
+          and message_status = 'needsMessage'
+          and is_opted_out = false
+      ),
+      campaign_summary_raw as (
+        select
+          c.id as campaign_id,
+          c.autosend_limit_max_contact_id,
+          count(cc.id) as total_count_to_send,
+          min(cc.id) as min_cc_id
+        from campaign c
+        left join sendable_contacts cc on cc.campaign_id = c.id
+        where true
+          -- organization requirements for autosending
+          and c.organization_id = $1
+          -- campaign requirements for autosending
+          and c.id = ANY ($2::integer[])
+          and c.is_archived = false
+          and c.is_started = true
+          and c.autosend_status = 'sending'
+        group by 1, 2
+      ),
+      campaign_summary as (
+        select
+          campaign_id,
+          (case
+            when total_count_to_send = 0 then 'complete'
+            when (
+              autosend_limit_max_contact_id is not null
+              and min_cc_id > autosend_limit_max_contact_id
+            ) then 'paused'
+            else null
+          end) as new_autosend_status
+        from campaign_summary_raw
+      )
+      update campaign
+      set autosend_status = new_autosend_status
+      from campaign_summary
       where true
-        -- organization requirements for autosending
-        and c.organization_id = $1
-        -- campaign requirements for autosending
-        and c.is_archived = false
-        and c.is_started = true
-        and c.autosend_status = 'sending'
-        and message_status = 'needsMessage'
-        and is_opted_out = false
-      group by 1
+        and organization_id = $1
+        and campaign_summary.campaign_id = campaign.id
+        and new_autosend_status is not null
```

Comment:
```suggestion
        and new_autosend_status is not null
        and id = ANY($2::integer[])
```
We only want to be able to mark campaigns as complete that we just queued, right? That's preserving previous behavior